src/client/QXmppMixManager.cpp

```diff
@@ -0,0 +1,1372 @@
+///
+/// \param channelJid JID of the channel
+///
+
+///
+/// Requests all participants of a MIX channel.
+///
+/// In the case of a channel that not everybody is allowed to participate in, the participants are a
+/// subset of the allowed JIDs.
+///
+/// \param channelJid JID of the channel
+///
+/// \return the result of the action
+///
+QXmppTask<QXmppMixManager::ParticipantResult> QXmppMixManager::requestParticipants(const QString &channelJid)
+{
+    return chain<ParticipantResult>(d->pubSubManager->requestItems<QXmppMixParticipantItem>(channelJid, ns_mix_node_participants.toString()), this, [](QXmppPubSubManager::ItemsResult<QXmppMixParticipantItem> &&result) -> ParticipantResult {
+        if (auto error = std::get_if<QXmppError>(&result)) {
+            return std::move(*error);
+        }
+
+        return std::get<QXmppPubSubManager::Items<QXmppMixParticipantItem>>(std::move(result)).items;
+    });
+}
+
+///
+/// \fn QXmppMixManager::userJoinedOrParticipantModified(const QString &channelJid, const QXmppMixParticipantItem &participantItem)
+///
+/// Emitted when a user joined a MIX channel or a participant changed the nick.
+///
+/// \param channelJid JID of the channel that is joined by the user or for which the participant changed the nick
+/// \param participantItem item for the new or modified participant
+///
+
+///
+/// \fn QXmppMixManager::participantLeft(const QString &channelJid, const QString &participantId)
+///
+/// Emitted when a participant left the MIX channel.
+///
+/// \param channelJid JID of the channel that is left by the participant
+/// \param participantId ID of the left participant
+///
+
+///
+/// Leaves a MIX channel.
+///
+/// \param channelJid JID of the channel to be left
+///
+/// \return the result of the action
+///
+QXmppTask<QXmppClient::EmptyResult> QXmppMixManager::leaveChannel(const QString &channelJid)
+{
+    QXmppMixIq iq;
+    iq.setType(QXmppIq::Set);
+    iq.setTo(client()->configuration().jidBare());
+    iq.setActionType(QXmppMixIq::ClientLeave);
+    iq.setChannelJid(channelJid);
+
+    return client()->sendGenericIq(std::move(iq));
+}
+
+///
+/// Deletes a MIX channel.
+///
+/// \param channelJid JID of the channel to be deleted
+///
+/// \return the result of the action
+///
+QXmppTask<QXmppClient::EmptyResult> QXmppMixManager::deleteChannel(const QString &channelJid)
+{
+    QXmppMixIq iq;
+    iq.setType(QXmppIq::Set);
+    iq.setTo(QXmppUtils::jidToDomain(channelJid));
+    iq.setActionType(QXmppMixIq::Destroy);
+    iq.setChannelId(QXmppUtils::jidToUser(channelJid));
+
+    return client()->sendGenericIq(std::move(iq));
+}
+
+///
+/// \fn QXmppMixManager::channelDeleted(const QString &channelJid)
+///
+/// Emitted when a MIX channel is deleted.
+///
+/// \param channelJid JID of the deleted channel
+///
+
+/// \cond
+void QXmppMixManager::onRegistered(QXmppClient *client)
+{
+    connect(client, &QXmppClient::disconnected, this, &QXmppMixManager::resetCachedData);
```

Comment:
Can you optimize this so it does cache the data if the stream was resumed? You can check the stream management state after (re-)connecting whether the last stream has been resumed.

https://doc.qxmpp.org/qxmpp-1/classQXmppClient.html#a65b66e4c1e2259a256bd6bd57d61bd57